doc/source/developing/testing.rst

```diff
@@ -284,8 +284,74 @@ Here is what a minimal example might look like for a new frontend:
 Answer test examples can be found in ``yt/frontends/enzo/tests/test_outputs.py``.
 
 
-How to Write Image Comparison Tests
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _update_image_tests:
+
+Creating and Updating Image Baselines for pytest-mpl Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We use `pytest-mpl <https://pytest-mpl.readthedocs.io/en/latest/>`_ for image comparison
+tests. These tests take the form of functions, which must be decorated with
+``@pytest.mark.mpl_image_compare`` and return a ``matplotlib.figure.Figure`` object.
+
+The collection of reference images is kept as git submodule in ``tests/pytest_mpl_baseline/``.
+
+There are 4 situations where updating reference images may be necessary
+
+- adding new tests
+- bugfixes
+- intentional change of style in yt
+- old baseline fails with a new version of matplotlib, but changes are not noticeable to the human eye
+
+The process of updating images is the same in all cases.
+
+1. open a Pull Request (PR) to yt's main repo with the code changes
+2. wait for tests jobs to complete
+3. go to the "Checks" tab on the PR page (``https://github.com/yt-project/yt/pull/<PR number>/checks``)
+4. if all tests passed, you're done !
+5. if tests other than image tests failed, fix them, and go back to step 2.
+  Otherwise, if only image tests failed, navigate to the "Build and Tests" job summary page.
+6. at the bottom of the page, you'll find "Artifacts".
+   Download ``yt_pytest_mpl_results.zip``, unzip it and open ``fig_comparison.html`` therein;
+   This document is an interactive report of the test job.
+   Inspect failed tests results and verify that any differences are either intended or insignificant.
+   If they are not, fix the code and go back to step 2
+7. clone ``https://github.com/yt-project/yt_pytest_mpl_baseline.git`` and unzip the new baseline
+8. Download the other artifact (``yt_pytest_mpl_new_baseline.zip``),
+   unzip it within your clone of ``yt_pytest_mpl_baseline``.
+9. create a branch, commit all changes, and open a PR to ``https://github.com/yt-project/yt_pytest_mpl_baseline``
+   (this PR should link to the one on the main repo)
+10. wait for this second PR to be merged
+11. go back to your local copy of ``yt``'s main repository.
+12. run the following commands
+
+.. code-block:: bash
+
+  $ git submodule update --init
+  $ cd tests/pytest_mpl_baseline
+  $ git checkout main
+  $ git pull
+  $ cd ../
+  $ git add pytest_mpl_baseline
+  $ git commit -m "update image test baseline"
```

Comment:
can we have this include the PR requiring the update?

Reply:
It says "update image baseline" -- but it would be really helpful to have it reference the yt PR it was updating as a result of.

Reply:
ah ! good call. I gave it a go rephrasing a little more than just this bullet point. Let me know it's any better.